Library/Homebrew/extend/pathname.rb

```diff
@@ -308,7 +308,7 @@ def compression_type
   # @private
   def elf?
     @which_file ||= which("file")
-    !@which_file.nil? && `#{@which_file} -b #{self}` =~ /^ELF/
+    !@which_file.nil? && `#{@which_file} -b #{self.to_s.gsub(/(\(|\))/,'\\\\\1')}` =~ /^ELF/
```

Comment:
Way better. Please rewrite this method to check whether the first four characters of the file are
```ruby
"\x7fELF"
```